### ISSUE TITLE
ProgrammingError: column "tags" does not exit, LINE 28: %#tags AS tags

### ISSUE THREAD

**Issue by jojoaddison** [mapzen/vector-datasource#804]
Traceback (most recent call last):
Following the following sql updates error,

 Applying updates in parallel across tables...
psql:apply-planet_osm_line.sql:64: ERROR:  record "row" has no field "tags"
LINE 31:         row.tags)
                 ^
QUERY:  SELECT mz_calculate_min_zoom_roads_(
        NULL,
        NULL,
        NULL,
        NULL,
        NULL,
        NULL,
        NULL,
        row."aerialway"::text,
        row."aeroway"::text,
        row."bicycle"::text,
        row."bridge"::text,
        row."foot"::text,
        row."highway"::text,
        row."horse"::text,
        row."layer"::text,
        row."leisure"::text,
        row."man_made"::text,
        row."oneway"::text,
        row."operator"::text,
        row."railway"::text,
        row."ref"::text,
        row."route"::text,
        row."service"::text,
        row."sport"::text,
        row."tracktype"::text,
        row."tunnel"::text,
        row.osm_id,
        row.name,
        row.way,
        row.tags)
CONTEXT:  PL/pgSQL function mz_calculate_min_zoom_roads(planet_osm_line) line 5 at RETURN
SQL statement "UPDATE planet_osm_line
  SET mz_road_level = mz_calculate_min_zoom_roads(planet_osm_line.*)
  WHERE mz_calculate_min_zoom_roads(planet_osm_line.*) IS NOT NULL"
PL/pgSQL function inline_code_block line 8 at SQL statement
psql:apply-planet_osm_point.sql:46: ERROR:  record "row" has no field "craft"
LINE 8:         row."craft",
                ^

A request to tileserver fails with the error below...

 File "/home/jack/src/tileserver/tileserver/__init__.py", line 190, in __call__
    response = self.handle_request(request)
  File "/home/jack/src/tileserver/tileserver/__init__.py", line 277, in handle_request
    feature_data_all = self.data_fetcher(coord, self.layer_config.all_layers)
  File "/home/jack/src/env/lib/python2.7/site-packages/tilequeue/query.py", line 195, in __call__
    raise async_exception
ProgrammingError: column "tags" does not exist
LINE 28:     %#tags AS tags
               ^
HINT:  There is a column named "tags" in table "*SELECT* 1", but it cannot be referenced from this part of the query.

This error occured after setting up a fresh installation of vector-datasource with tileserver.

Any assistance will be greatly appreciated.

**Comment by rmarianski** [comment]
Did you import a pbf with the [vector-datasource style file](https://github.com/mapzen/vector-datasource/blob/784f22a1b8b230ed0c57c98cf860bf5939fc56c9/osm2pgsql.style) and with the hstore flag? It should look like: https://github.com/mapzen/vector-datasource/wiki/Mapzen-Vector-Tile-Service#load-pbf-data

I tried a fresh import with a small extract and the sql updates worked for me. Do you mind trying with a new database again? Immediately after the osm2pgsql pbf import step, you should have `craft` and `tags` columns in the `planet_osm_polygon`, `planet_osm_line`, and `planet_osm_point` tables.

**Comment by jojoaddison** [comment]
Hello @rmarianski,
Thanks for your prompt response.
I already have the whole planet imported into postgis. 
How would you advise I go about it?
Do I have to extend the the planet_xxx tables with the columns and try updating?
Thanks for your assistance.

**Comment by rmarianski** [comment]
I'd recommend re-importing it again using the style file linked to above and in the docs. This will take a while with the whole planet, so you might want to try the process with a [metro extract](https://mapzen.com/data/metro-extracts/) first to verify that it works for you.

Our updates and queries expect the data imported in a certain way, specifically with the osm2pgsql style in the vector-datasource repo. If you've imported the data a different way, you'll get failures the first time anything references a column that doesn't exist.

**Comment by nvkelso** [comment]
Try importing a small Metro Extract to verify in a few minutes instead of hours for a new planet import. If that works, then do the whole planet.

https://mapzen.com/data/metro-extracts/

**Comment by jojoaddison** [comment]
Hi Guys,
Thanks for the support.
I have successfully imported the data but still can't tileserver to work.
I will close this thread for now and post the other one in a new issue.
Thanks.
